File: Ejercicio_03/Domain/Interfaces/Departamento/IDepartamentoRepository.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Domain.Entities;

namespace Domain.Interfaces
{
    public interface IDepartamentoRepository
    {
        List<Departamento> getDepartamentos();
        Departamento getDepartamento(int id);
        int deleteDepartamento(int id);
        int updateDepartamento(int id, Departamento departamento);
        int addDepartamento(Departamento departamento);
    }
}

File: Angular/CRUD/src/domain/interfaces/IDepartamentoRepository.ts
import { Departamento } from "../entities/Departamento"

export interface IDepartamentoRepository {
    getDepartamentos(): Promise<Array<Departamento>>
	getDepartamento(id: number): Promise<Departamento>
	updateDepartamento(id: number, departamento: Departamento): Promise<number>
	addDepartamento(departamento: Departamento): Promise<number>
	deleteDepartamento(id: number): Promise<number>
}
